drake/examples/kuka_iiwa_arm/iiwa_controller.cc

```diff
@@ -0,0 +1,173 @@
+                  plan_source->get_plan_input_port());
+  builder.Connect(status_sub->get_output_port(0),
+                  plan_source->get_status_input_port());
+
+  builder.Connect(status_sub->get_output_port(0),
+                  status_receiver->get_input_port(0));
+
+  Eigen::VectorXd Kp = Eigen::VectorXd::Zero(kNumJoints);
+  Eigen::VectorXd Ki = Eigen::VectorXd::Ones(kNumJoints) * 0.05;
+  Eigen::VectorXd Kd = Eigen::VectorXd::Zero(kNumJoints);
+
+  auto control_ports = systems::PidControlledSystem<double>::ConnectController(
+      command_sender->get_input_port(0),
+      status_receiver->get_measured_position_output_port(), nullptr, Kp, Ki, Kd,
+      &builder);
+  builder.Connect(plan_source->get_output_port(0),
+                  target_demux->get_input_port(0));
+  builder.Connect(target_demux->get_output_port(0),
+                  control_ports.control_input_port);
+  builder.Connect(plan_source->get_output_port(0),
+                  control_ports.state_input_port);
+  builder.Connect(command_sender->get_output_port(0),
+                  command_pub->get_input_port(0));
+  auto diagram = builder.Build();
+
+  auto context = diagram->CreateDefaultContext();
+  auto output = diagram->AllocateOutput(*context);
+  std::unique_ptr<systems::DiscreteState<double>> discrete_state =
+      diagram->AllocateDiscreteVariables();
+  std::unique_ptr<systems::State<double>> tmp_state = context->CloneState();
+
+  lcm.StartReceiveThread();
+  drake::log()->info("controller started");
+
+  // TODO(sam.creasey) All of the remaining code in this function is
+  // basically duplicative of a similar implementation in
+  // valkyrie_balancing_controller_system.cc, and I think they could
+  // easily be merged into a single function templated on the message
+  // type.  Do this once we get to rule-of-three.
+
+  // Loops until the first status message arrives.
+  while (true) {
+    // Sets Context's time to the timestamp in the bot_core::robot_state_t msg.
+    const lcmt_iiwa_status* msg =
+        status_receiver->EvalInputValue<lcmt_iiwa_status>(
+            diagram->GetSubsystemContext(*context, status_receiver), 0);
+    context->set_time(static_cast<double>(msg->utime) / 1e6);
+    if (context->get_time() != 0) break;
+  }
+  drake::log()->info("status received");
+
+  // Create an integrator to advance the state of the integrator
+  // portion of the PidControlledSystem above.  A basic euler
+  // integrator should do, precision is not critical for this
+  // application.
+  systems::ExplicitEulerIntegrator<double> integrator(
+      *diagram, kIiwaLcmStatusPeriod, context.get());
+  integrator.Initialize();
+
+  systems::UpdateActions<double> update_actions;
+  double next_control_time =
+      diagram->CalcNextUpdateTime(*context, &update_actions);
+
+  // Loop forever, continuing to update the time based on the incoming
+  // status message.
+  while (true) {
+    if (next_control_time <= context->get_time()) {
```

Comment:
a small suggestion, I think that for improved readability you could put the busy wait loop at the top, and then remove the `if` conditional so there isn't any nested code.

```
while (true) {

  while (busy wait condition) { }

   // do control

}
```